src/rgw/rgw_sal_motr.cc

```diff
@@ -443,6 +449,25 @@ int MotrUser::store_user(const DoutPrefixProvider* dpp,
     secret_key = k.key;
     MotrAccessKey MGWUserKeys(access_key, secret_key, info.user_id.to_str());
     store->store_access_key(dpp, y, MGWUserKeys);
+    access_key_tracker.insert(access_key);
+  }
+
+  // Check if any key need to be deleted
```

Comment:
Not clear on why we would end up in deleting a key while storing a user ?

Reply:
We will be deleting the key in _delete key_ operation only.
In case of rados and dbstore, access key is stored with user info only. But in case of motr, apart from user info, we also store access key in access key index table. When key deletion is requested, RGW takes care of deleting the key from user info. But we need to explicitly delete it from our access key index table. 

Now motr does not talk to user requests directly, it comes through sal layer. So, we don't have a way to get request attributes in motr sal code. When a key is deleted, the updated user info needs to be stored and that's where store_user() function is called. 

To identify which key needs to be deleted, we keep an access key tracker. In case of deletion there will be mismatch between number of keys user info have and number of keys tracked. If that's the case, we delete the mismatched key from access key index table.